setup.py

```diff
@@ -179,18 +179,14 @@
     )]
 else:
     print('Using cffi extension.')
-    install_requires = ['cffi>=0.8']
-    if platform.python_implementation() == 'PyPy':
-        print('Using cffi with PyPy, no extension module to build.')
+    install_requires = ['cffi>=0.8; implementation_name=="cpython"']
```

Comment:
this requirement should be unconditional, rather than being in an `else` branch.  Better for solvers like uv, poetry etc to have the same metadata in all wheels